### DISCUSSION TITLE
Allow custom concurrent PR limit per package file

### DISCUSSION THREAD

**Discussion by allanlewis** [renovatebot/renovate#24438] [Suggest an Idea]
### Tell us more.

As per #24420, it isn't currently possible to limit the number of PRs raised per package file. This might be desirable as PRs affecting a given package file (e.g. a `pyproject.toml` or `package.json`) will inevitably conflict, so we might only want one PR open per package file.

In my case, I have a monorepo with ~30 package files and I would like to allow one open PR per package file but a higher limit overall.

**Comment by rarkins** [comment]
Please rewrite your title/description to use Renovate terminology (e.g. "package file" instead of "spec") and use it in the same way Renovate does. I suspect you are using "group" with a different meaning than Renovate uses. In Renovate terminology, updates are grouped into a single PR, so asking for a limit per group doesn't make sense.

**Reply by allanlewis** [reply]
> Please rewrite your title/description to use Renovate terminology (e.g. "package file" instead of "spec") and use it in the same way Renovate does.

I've made some edits, hopefully that reads better now?

> I suspect you are using "group" with a different meaning than Renovate uses.

I'm using it to mean "one entry in `packageRules`". Given each of those has a "group name" I've called that concept a "group": if that's not correct I'd be happy to rephrase.

**Reply by rarkins** [reply]
Is there a reason why you still write 'per group' instead of 'per package file'?

**Reply by allanlewis** [reply]
I've replaced the remaining instance of "group" with "package file". The reason I said "group" is that I was assuming the way to handle this would be to set `prConcurrentLimit` in a `packageRules` entry, but perhaps I shouldn't assume that.

**Comment by rarkins** [comment]
I've copy pasted your requirement into existing feature request #10546 as it should be covered as part of that rather than needing a separate feature

**Reply by allanlewis** [reply]
Thanks! I'll follow that issue.

**Comment by allanlewis** [comment]
Closing as #10546 has been resolved.
